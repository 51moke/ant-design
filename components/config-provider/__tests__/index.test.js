import React from 'react';
import { render } from 'enzyme';
import moment from 'moment';
import ConfigProvider from '..';
import Alert from '../../alert';
import Anchor from '../../anchor';
import AutoComplete from '../../auto-complete';
import Avatar from '../../avatar';
import BackTop from '../../back-top';
import Badge from '../../badge';
import Breadcrumb from '../../breadcrumb';
import Button from '../../button';
import Calendar from '../../calendar';
import Card from '../../card';
import Carousel from '../../carousel';
import Cascader from '../../cascader';
import Checkbox from '../../checkbox';
import Collapse from '../../collapse';
import Comment from '../../comment';
import DatePicker from '../../date-picker';
import Divider from '../../divider';
import Drawer from '../../drawer';
import Dropdown from '../../dropdown';
import Form from '../../form';
import { Row, Col } from '../../grid';

import Menu from '../../menu';
import Input from '../../input';

describe('ConfigProvider', () => {
  describe('components', () => {
    function testPair(name, renderComponent) {
      describe(name, () => {
        // normal
        it('normal', () => {
          expect(render(renderComponent({}))).toMatchSnapshot();
        });

        // prefixCls
        it('prefixCls', () => {
          expect(render(renderComponent({ prefixCls: `prefix-${name}` }))).toMatchSnapshot();
        });

        // configProvider
        it('configProvider', () => {
          expect(render(
            <ConfigProvider prefixCls="config">
              {renderComponent({})}
            </ConfigProvider>
          )).toMatchSnapshot();
        });
      });
    }

    // Alert
    testPair('Alert', props => (
      <Alert {...props} message="Bamboo is Little Light" type="success" />
    ));

    // Anchor
    testPair('Anchor', props => (
      <Anchor {...props}>
        <Anchor.Link {...props} href="#bamboo" title="Little Light" />
      </Anchor>
    ));

    // AutoComplete
    testPair('AutoComplete', props => (
      <AutoComplete {...props} />
    ));

    // Avatar
    testPair('Avatar', props => (
      <Avatar {...props} />
    ));

    // BackTop
    testPair('BackTop', props => (
      <BackTop visible {...props} />
    ));

    // Badge
    testPair('Badge', (props) => {
      const newProps = {
        ...props,
      };

      // Hook for additional `scrollNumberPrefixCls` prop
      if (props.prefixCls) {
        newProps.scrollNumberPrefixCls = 'prefix-scroll-number';
      }

      return (
        <div>
          <Badge {...newProps} count={5}>
            <span />
          </Badge>

          <Badge {...newProps} dot>
            <span />
          </Badge>
        </div>
      );
    });

    // Breadcrumb
    testPair('Breadcrumb', props => (
      <Breadcrumb {...props}>
        <Breadcrumb.Item {...props}>Bamboo</Breadcrumb.Item>
        <Breadcrumb.Item {...props}>Light</Breadcrumb.Item>
      </Breadcrumb>
    ));

    // Button
    testPair('Button', props => (
      <div>
        <Button {...props}>Bamboo</Button>
        <Button.Group {...props}>
          <Button {...props}>Little</Button>
          <Button {...props}>Light</Button>
        </Button.Group>
      </div>
    ));

    // Calendar
    testPair('Calendar', props => (
      <div>
        <Calendar {...props} value={moment('2000-09-03')} mode="month" />
        <Calendar {...props} value={moment('2000-09-03')} mode="year" />
      </div>
    ));

    // Card
    testPair('Card', props => (
      <Card {...props}>
        <Card.Grid {...props}>
          <Card.Meta {...props} />
        </Card.Grid>
      </Card>
    ));

    // Carousel
    testPair('Carousel', props => (
      <Carousel {...props}>
        <div><h3>Bamboo</h3></div>
        <div><h3>Light</h3></div>
      </Carousel>
    ));

    // Cascader
    testPair('Cascader', props => (
      <Cascader {...props} options={[]} showSearch />
    ));

    // Checkbox
    testPair('Checkbox', props => (
      <Checkbox.Group {...props}>
        <Checkbox {...props}>Bamboo</Checkbox>
      </Checkbox.Group>
    ));

    // Collapse
    testPair('Collapse', props => (
      <Collapse {...props}>
        <Collapse.Panel header="Bamboo">
          <p>Light</p>
        </Collapse.Panel>
      </Collapse>
    ));

    // Comment
    testPair('Comment', props => (
      <Comment {...props} content="Bamboo">
        <Comment {...props} content="Light" />
      </Comment>
    ));

    // DatePicker
    describe('DatePicker', () => {
      testPair('DatePicker', props => (
        <div>
          <DatePicker {...props} />
        </div>
      ));

      // RangePicker
      testPair('RangePicker', props => (
        <div>
          <DatePicker.RangePicker {...props} />
        </div>
      ));

      // MonthPicker
      testPair('MonthPicker', props => (
        <div>
          <DatePicker.MonthPicker {...props} />
        </div>
      ));

      // WeekPicker
      testPair('WeekPicker', props => (
        <div>
          <DatePicker.WeekPicker {...props} />
        </div>
      ));
    });

    // Divider
    testPair('Divider', props => (
      <Divider {...props} />
    ));

    // Drawer
    testPair('Drawer', props => (
      <Drawer {...props} visible getContainer={false} />
    ));

    // Dropdown
    testPair('Dropdown', (props) => {
      const menu = (
        <Menu {...props}>
          <Menu.Item {...props}>Bamboo</Menu.Item>
        </Menu>
      );

      return (
        <Dropdown.Button {...props} overlay={menu}>
          Light
        </Dropdown.Button>
      );
    });

    // Form
    testPair('Form', props => (
      <Form {...props}>
        <Form.Item
          {...props}
          validateStatus="error"
          help="Bamboo is Light"
        >
          <Input {...props} />
        </Form.Item>
      </Form>
    ));

    // Grid
    testPair('Grid', (props) => {
      const rowProps = {};
      const colProps = {};
      if (props.prefixCls) {
        rowProps.prefixCls = 'prefix-row';
        colProps.prefixCls = 'prefix-col';
      }

      return (
        <Row {...rowProps}>
          <Col {...colProps} span={1} />
        </Row>
      );
    });
  });
});
