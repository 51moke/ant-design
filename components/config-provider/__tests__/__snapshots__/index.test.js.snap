// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigProvider components alert configProvider 1`] = `
<div
  class="config-alert config-alert-success config-alert-no-icon"
  data-show="true"
>
  <span
    class="config-alert-message"
  >
    Bamboo is Little Light
  </span>
  <span
    class="config-alert-description"
  />
</div>
`;

exports[`ConfigProvider components alert normal 1`] = `
<div
  class="ant-alert ant-alert-success ant-alert-no-icon"
  data-show="true"
>
  <span
    class="ant-alert-message"
  >
    Bamboo is Little Light
  </span>
  <span
    class="ant-alert-description"
  />
</div>
`;

exports[`ConfigProvider components alert prefixCls 1`] = `
<div
  class="prefix-alert prefix-alert-success prefix-alert-no-icon"
  data-show="true"
>
  <span
    class="prefix-alert-message"
  >
    Bamboo is Little Light
  </span>
  <span
    class="prefix-alert-description"
  />
</div>
`;

exports[`ConfigProvider components anchor configProvider 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="config-anchor-wrapper"
      style="max-height:100vh"
    >
      <div
        class="config-anchor"
      >
        <div
          class="config-anchor-ink"
        >
          <span
            class="config-anchor-ink-ball"
          />
        </div>
        <div
          class="config-anchor-link"
        >
          <a
            class="config-anchor-link-title"
            href="#bamboo"
            title="Little Light"
          >
            Little Light
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfigProvider components anchor normal 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="ant-anchor-wrapper"
      style="max-height:100vh"
    >
      <div
        class="ant-anchor"
      >
        <div
          class="ant-anchor-ink"
        >
          <span
            class="ant-anchor-ink-ball"
          />
        </div>
        <div
          class="ant-anchor-link"
        >
          <a
            class="ant-anchor-link-title"
            href="#bamboo"
            title="Little Light"
          >
            Little Light
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfigProvider components anchor prefixCls 1`] = `
<div>
  <div
    class=""
  >
    <div
      class="prefix-anchor-wrapper"
      style="max-height:100vh"
    >
      <div
        class="prefix-anchor"
      >
        <div
          class="prefix-anchor-ink"
        >
          <span
            class="prefix-anchor-ink-ball"
          />
        </div>
        <div
          class="prefix-anchor-link"
        >
          <a
            class="prefix-anchor-link-title"
            href="#bamboo"
            title="Little Light"
          >
            Little Light
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ConfigProvider components auto-complete configProvider 1`] = `
<div
  class="config-select-show-search config-select-auto-complete ant-select ant-select-combobox ant-select-enabled"
>
  <div
    aria-autocomplete="list"
    aria-controls="test-uuid"
    aria-expanded="false"
    aria-haspopup="true"
    class="ant-select-selection
            ant-select-selection--single"
    role="combobox"
  >
    <div
      class="ant-select-selection__rendered"
    >
      <ul>
        <li
          class="ant-select-search ant-select-search--inline"
        >
          <div
            class="ant-select-search__field__wrap"
          >
            <input
              class="ant-input ant-select-search__field"
              type="text"
              value=""
            />
            <span
              class="ant-select-search__field__mirror"
            >
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <span
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <i
        class="anticon anticon-down ant-select-arrow-icon"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="down"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </i>
    </span>
  </div>
</div>
`;

exports[`ConfigProvider components auto-complete normal 1`] = `
<div
  class="ant-select-show-search ant-select-auto-complete ant-select ant-select-combobox ant-select-enabled"
>
  <div
    aria-autocomplete="list"
    aria-controls="test-uuid"
    aria-expanded="false"
    aria-haspopup="true"
    class="ant-select-selection
            ant-select-selection--single"
    role="combobox"
  >
    <div
      class="ant-select-selection__rendered"
    >
      <ul>
        <li
          class="ant-select-search ant-select-search--inline"
        >
          <div
            class="ant-select-search__field__wrap"
          >
            <input
              class="ant-input ant-select-search__field"
              type="text"
              value=""
            />
            <span
              class="ant-select-search__field__mirror"
            >
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <span
      class="ant-select-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <i
        class="anticon anticon-down ant-select-arrow-icon"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="down"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </i>
    </span>
  </div>
</div>
`;

exports[`ConfigProvider components auto-complete prefixCls 1`] = `
<div
  class="prefix-auto-complete-show-search prefix-auto-complete-auto-complete prefix-auto-complete prefix-auto-complete-combobox prefix-auto-complete-enabled"
>
  <div
    aria-autocomplete="list"
    aria-controls="test-uuid"
    aria-expanded="false"
    aria-haspopup="true"
    class="prefix-auto-complete-selection
            prefix-auto-complete-selection--single"
    role="combobox"
  >
    <div
      class="prefix-auto-complete-selection__rendered"
    >
      <ul>
        <li
          class="prefix-auto-complete-search prefix-auto-complete-search--inline"
        >
          <div
            class="prefix-auto-complete-search__field__wrap"
          >
            <input
              class="ant-input prefix-auto-complete-search__field"
              type="text"
              value=""
            />
            <span
              class="prefix-auto-complete-search__field__mirror"
            >
               
            </span>
          </div>
        </li>
      </ul>
    </div>
    <span
      class="prefix-auto-complete-arrow"
      style="user-select:none;-webkit-user-select:none"
      unselectable="on"
    >
      <i
        class="anticon anticon-down prefix-auto-complete-arrow-icon"
      >
        <svg
          aria-hidden="true"
          class=""
          data-icon="down"
          fill="currentColor"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
          />
        </svg>
      </i>
    </span>
  </div>
</div>
`;

exports[`ConfigProvider components avatar configProvider 1`] = `
<span
  class="config-avatar config-avatar-circle"
>
  <span
    class="config-avatar-string"
  />
</span>
`;

exports[`ConfigProvider components avatar normal 1`] = `
<span
  class="ant-avatar ant-avatar-circle"
>
  <span
    class="ant-avatar-string"
  />
</span>
`;

exports[`ConfigProvider components avatar prefixCls 1`] = `
<span
  class="prefix-avatar prefix-avatar-circle"
>
  <span
    class="prefix-avatar-string"
  />
</span>
`;

exports[`ConfigProvider components back-top configProvider 1`] = `
<div
  class="config-back-top"
>
  <div
    class="config-back-top-content"
  >
    <div
      class="config-back-top-icon"
    />
  </div>
</div>
`;

exports[`ConfigProvider components back-top normal 1`] = `
<div
  class="ant-back-top"
>
  <div
    class="ant-back-top-content"
  >
    <div
      class="ant-back-top-icon"
    />
  </div>
</div>
`;

exports[`ConfigProvider components back-top prefixCls 1`] = `
<div
  class="prefix-back-top"
>
  <div
    class="prefix-back-top-content"
  >
    <div
      class="prefix-back-top-icon"
    />
  </div>
</div>
`;
